### PR TITLE
Implement fixes and changes based on tests

### PR DESCRIPTION
**Pull request by denessapi**

### CODE REVIEW

File: trap/src/main/java/com/cursorinsight/trap/transport/TrapWebsocketTransport.kt [hunk 1]
```diff
@@ -46,7 +46,7 @@ internal class TrapWebsocketTransport: TrapTransport {
         websocket = null
     }
 
-    override fun send(data: String) {
+     override fun send(data: String, avoidSendingTooMuchData: Boolean) {
```
Comment (gbence) [inline trap/src/main/java/com/cursorinsight/trap/transport/TrapWebsocketTransport.kt]: Indentation

File: trap/src/test/java/com/cursorinsight/trap/transport/TrapHttpTransportTest.kt [hunk 2]
```diff
@@ -74,7 +74,7 @@ class TrapHttpTransportTest {
                 readTimeout = 100,
                 compress = false,
             ))
-        transport.send("[[999]]")
+        transport.send("[[999]]", false)
```
Comment (gbence) [inline trap/src/test/java/com/cursorinsight/trap/transport/TrapHttpTransportTest.kt]: How about a `send()` method with an optional second argument that defaults to `false`?

File: trap/src/main/java/com/cursorinsight/trap/TrapConfig.kt [hunk 3]
```diff
@@ -65,14 +65,19 @@ data class TrapConfig(
     /**
      * The size of the circular data queue.
      */
-    var queueSize: Int = 2048,
+     var queueSize: Int = 8192,
```
Comment (gbence) [inline trap/src/main/java/com/cursorinsight/trap/TrapConfig.kt]: Indentation

File: trap/src/main/java/com/cursorinsight/trap/TrapManager.kt [hunk 4]
```diff
@@ -270,14 +249,38 @@ class TrapManager internal constructor(
         }
     }
 
+    private fun getLowBatteryStatus() : Boolean {
```
Comment (gbence) [inline trap/src/main/java/com/cursorinsight/trap/TrapManager.kt:260]: Does it make sense to share code between this function and TrapBatteryCollector's `processBatteryStatus()`?

**Review by gbence — CHANGES_REQUESTED** [review]

File: trap/src/test/java/com/cursorinsight/trap/transport/TrapHttpTransportTest.kt (hunk 2 above)
Comment (denessapi) [reply trap/src/test/java/com/cursorinsight/trap/transport/TrapHttpTransportTest.kt]: Done

File: trap/src/main/java/com/cursorinsight/trap/transport/TrapWebsocketTransport.kt (hunk 1 above)
Comment (denessapi) [reply trap/src/main/java/com/cursorinsight/trap/transport/TrapWebsocketTransport.kt]: Done

File: trap/src/main/java/com/cursorinsight/trap/TrapConfig.kt (hunk 3 above)
Comment (denessapi) [reply trap/src/main/java/com/cursorinsight/trap/TrapConfig.kt]: Done

File: trap/src/main/java/com/cursorinsight/trap/TrapManager.kt (hunk 4 above)
Comment (denessapi) [reply trap/src/main/java/com/cursorinsight/trap/TrapManager.kt:260]: As we are subscribing on different events in the Manager and Collector and the collector reads more properties from the intent I don't see any easy way to do it.

Comment (gbence) [reply trap/src/main/java/com/cursorinsight/trap/TrapManager.kt:260]: :+1:

**Review by gbence — APPROVED** [review]

**Pull request merged**